src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Refactoring/RazorComponentRenameEndpointTest.cs

```diff
@@ -94,6 +96,65 @@ public async Task Handle_Rename_WithNamespaceDirective()
             Assert.Equal(24, editChangeEdit2.Range.End.Character);
         }
 
+        [Fact]
+        public async Task Handle_Rename_FullyQualifiedAndNot()
+        {
+            // Arrange
+            var request = new RenameParams
+            {
+                TextDocument = new TextDocumentIdentifier
+                {
+                    Uri = new Uri("file:///c:/First/Index.razor")
+                },
+                Position = new Position(2, 1),
+                NewName = "Component5"
+            };
+
+            // Act
+            var result = await _endpoint.Handle(request, CancellationToken.None);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(3, result.DocumentChanges.Count());
+            var renameChange = result.DocumentChanges.ElementAt(0);
+            Assert.True(renameChange.IsRenameFile);
+            Assert.Equal("file:///c:/First/Component1337.razor", renameChange.RenameFile.OldUri);
+            Assert.Equal("file:///c:/First/Component5.razor", renameChange.RenameFile.NewUri);
+            var editChange1 = result.DocumentChanges.ElementAt(1);
+            Assert.True(editChange1.IsTextDocumentEdit);
+            Assert.Equal("file:///c:/First/Index.razor", editChange1.TextDocumentEdit.TextDocument.Uri.ToString());
+            Assert.Equal(2, editChange1.TextDocumentEdit.Edits.Count());
+            var editChange1Edit1 = editChange1.TextDocumentEdit.Edits.ElementAt(0);
+            Assert.Equal("Component5", editChange1Edit1.NewText);
```

Comment:
I feel like this might be more readable as something like:
```
Assert.Collection((edit) =>{ Assert.Equal(new EditChange{...}, edit);})
```

But that's just personal preference and I see that you're just matching the style in the rest of these tests.

Reply:
Can do